src/Microsoft.DotNet.ProjectModel/Graph/LockFile.cs

```diff
@@ -87,5 +84,18 @@ public bool IsValidForProject(Project project, out string message)
             message = null;
             return true;
         }
+
+        public void MergeWith(LockFile fragmentLockFile)
+        {
+            if (Version != fragmentLockFile.Version)
+            {
+                throw new ArgumentException($"Cannot merge master lock file {LockFilePath} with fragment lock file {fragmentLockFile.LockFilePath} due to missmatching versions");
+            }
+
+            Targets.MergeWith(fragmentLockFile.Targets, l => l.TargetFramework.Framework + l.RuntimeIdentifier);
+            ProjectLibraries.MergeWith(fragmentLockFile.ProjectLibraries, pl => pl.Name + pl.Version);
+            PackageLibraries.MergeWith(fragmentLockFile.PackageLibraries, pl => pl.Name + pl.Version);
```

Comment:
Same here. Make the key match the Lock File property name by putting a `/` between Name and Version